### DISCUSSION TITLE
Making player jump on collision

### DISCUSSION THREAD

**Discussion by weirdyang** [replit/kaboom#676] [Q&A]
I’m trying to build a platformer where the player will automatically jump on collision with floor

~~~
player.onCollide("floor", ()=> {
	debug.log("hit");
	player.jump();
})
~~~

Why doesn’t this work?

**Comment by slmjkdbtl** [comment]
This works in [v3000](https://3000.kaboomjs.com/blog/3000)

**Reply by weirdyang** [reply]
Oh so I can’t do this in v2?

**Reply by slmjkdbtl** [reply]
Yeah sorry v2's physics system has some restrictions, it'll be quite some work to port the new physics system to v2 to fix this, currently the support for v2 is mainly for small scaled bug fixes as we focus more on v3
